sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.VisitStmt.cs

```diff
@@ -41,7 +41,32 @@ private void VisitBinaryOperator(BinaryOperator binaryOperator)
         outputBuilder.Write(' ');
         outputBuilder.Write(binaryOperator.OpcodeStr);
         outputBuilder.Write(' ');
-        Visit(binaryOperator.RHS);
+
+        if (binaryOperator.IsShiftOp || binaryOperator.IsShiftAssignOp)
+        {
+            // RHS of shift operation in C# must be an int
+            if (binaryOperator.RHS.Type.Kind is CXType_Int or CXType_Long)
+            {
+                Visit(binaryOperator.RHS);
+            }
+            else if (binaryOperator.RHS is IntegerLiteral literal)
+            {
+                outputBuilder.Write(literal.Value);
+            }
```

Comment:
This won't do the right thing for an integer literal whose value is actually larger than 32-bits. Which while unlikely, should probably be handled.